ansible_base/lib/utils/encryption.py

```diff
@@ -70,3 +79,26 @@ def decrypt_string(self, value: str) -> str:
 
 
 ansible_encryption = SimpleLazyObject(func=lambda: Fernet256())
+
+
+def generate_hmac_sha256_shared_secret(nonce: Optional[str] = None) -> str:
+    shared_secret = getattr(settings, 'ANSIBLE_BASE_SHARED_SECRET', None)
+    if shared_secret is None:
+        logger.error("The setting ANSIBLE_BASE_SHARED_SECRET was not set, insecurely using default")
+        shared_secret = 'ansible_base_shared_secret_insecure'
```

Comment:
I really don't like defaulting to a well-known PSK. That seems like it is going to get us a CVE.

Reply:
I made it raise an exception and will just not validate/generate the headers if this exception is thrown. For other exceptions I am doing to do a `logger.exectpion`.